gremlin-go/driver/connection_test.go

```diff
@@ -7,44 +7,49 @@ import (
 	"golang.org/x/text/language"
 )
 
-const runIntegration = false
+const runIntegration = true
 
 func TestConnection(t *testing.T) {
-	t.Run("Test connect", func(t *testing.T) {
+	t.Run("Test createConnection", func(t *testing.T) {
 		if runIntegration {
-			connection := connection{"localhost", 8181, Gorilla, newLogHandler(&defaultLogger{}, Info, language.English), nil, nil, nil}
-			err := connection.connect()
+			connection, err := createConnection(newLogHandler(&defaultLogger{}, Info, language.English), "localhost", 8181)
+			assert.Nil(t, err)
+			assert.NotNil(t, connection)
+			err = connection.close()
 			assert.Nil(t, err)
 		}
 	})
 
 	t.Run("Test write", func(t *testing.T) {
 		if runIntegration {
-			connection := connection{"localhost", 8181, Gorilla, newLogHandler(&defaultLogger{}, Info, language.English), nil, nil, nil}
-			err := connection.connect()
+			connection, err := createConnection(newLogHandler(&defaultLogger{}, Info, language.English), "localhost", 8181)
 			assert.Nil(t, err)
+			assert.NotNil(t, connection)
 			request := makeStringRequest("g.V().count()")
 			resultSet, err := connection.write(&request)
 			assert.Nil(t, err)
 			assert.NotNil(t, resultSet)
 			result := resultSet.one()
 			assert.NotNil(t, result)
 			assert.Equal(t, "[0]", result.GetString())
+			err = connection.close()
+			assert.Nil(t, err)
 		}
 	})
 
 	t.Run("Test client submit", func(t *testing.T) {
 		if runIntegration {
-			connection := connection{"localhost", 8181, Gorilla, newLogHandler(&defaultLogger{}, Info, language.English), nil, nil, nil}
-			err := connection.connect()
+			client, err := NewClient("localhost", 8181)
```

Comment:
- Can we update documentation (how to prepare an environment for Integration tests?)

Perhaps can be done with
```
docker pull tinkerpop/gremlin-server
docker run -d -p 8182:8182  --name gremlin-server tinkerpop/gremlin-server
```

 - Would it be better to use port 8182 since it is a default gremlin-server port? Can we extract the port to a const (so it can be changed only in one place if tests have to be run against some specific server?)

Reply:
Good idea, will look into this.

Reply:
Created ticket to handle this.